addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1848,6 +1848,17 @@ var SnippetsMenu = Widget.extend({
                 }
 
                 return editorToEnable;
+            }).then(editor => {
+                if (this._currentTab === this.tabs.OPTIONS &&
+                    this.options.wysiwyg.linkTools &&
+                    this.options.wysiwyg.linkTools.el
+                ) {
+                    this.options.wysiwyg.linkTools.el.scrollIntoView();
+                    if (!this.options.wysiwyg.linkTools.noFocusUrl) {
+                        this.options.wysiwyg.linkTools.focusUrl();
+                    }
+                }
+                return editor;
             });
```

Comment:
Better than setTimeout and/or a internal-loop... but I am not sure to understand: does not this mean that any click on a snippet would focus the URL tools? (as done at each snippet activation?) :thinking: 

Also:
- Why a second .then for this? Can't it just be before the `return editorToEnable`?
- Needs comments
- Should not the scrollIntoView be part of the focusURL method?
- Not required but FWI how I would write those big if (logic operator first and different indentaion) (no need to change that if you prefer like you did :+1:):
```javascript
                if (this._currentTab === this.tabs.OPTIONS
                        && this.options.wysiwyg.linkTools
                        && this.options.wysiwyg.linkTools.el) {
                    this.stuff();
                }
```

Reply:
The second `then` was needed because the first one can lead to an early return if the editor to enable is already displayed, it was needed in order to focus the url when double clicking on two different links of the same snippet.
I moved the scrollIntoView inside the focusUrl method, I had left it out because I thought that we wanted to scroll down to the link tools every time they were shown, wether or not the url input should be focused

Reply:
The check on the tab and for the linktools will be performed at each _activateSnippet, but the url will be focused only in the case of a click on a link, as a click anywhere else will destroy and remove the link tools

Reply:
> I had left it out because I thought that we wanted to scroll down to the link tools every time they were shown, wether or not the url input should be focused

Mmh, maybe that's what we want. Although I think the final result here is fine, if the user wants to edit he can double click or use the tooltip actions and it is explicit enough then :+1: